### ISSUE TITLE
towed rolling stocks don't forward rolling resistance values in the right unit

### ISSUE THREAD

**Issue by eckter** [OpenRailAssociation/osrd#9657]
### What happened?

When sending an stdcm request for a given rolling stock with no towed rolling stock, the rolling resistance have this value (in the payload editoast -> core):

```json
        "rolling_resistance": {
            "type": "davis",
            "A": 2200.0,
            "B": 64.8,
            "C": 5.832
        },
```

Same rolling stock but with a towed rolling stock:

```json
        "rolling_resistance": {
            "type": "davis",
            "A": 2200000.0,
            "B": 233280.0,
            "C": 75582.72
        },
```

Unsurprisingly, we always fail to find a solution. This happens for any value in the towed rolling stock data. 

### What did you expect to happen?

Values with the same order of magnitude as before

### How can we reproduce it (as minimally and precisely as possible)?

Run any kind of request with a towed rolling stock (and weight and such otherwise it's ignored). Check the payloads received by core (can be done by setting `LOG_STDCM_REQUESTS=true`

### On which environments the bug occurs?

Local, Recette (SNCF), Dev (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

8461f3efed951b9b360c4039b9ae5d4adc60136c

**Comment by woshilapin** [comment]
TLDR: we might have some unit problems combined with incorrect formulas (mostly due to incorrect units).

Our original data for rolling resistance on traction engines are in the following units:
- solid friction: daN
- viscosity friction: daN/(km/h)
- aerodynamic friction: daN/(km/h)²

Then we use our internal tool to convert these into [International System of Unit (SI)](https://www.wikiwand.com/en/articles/International_System_of_Units) which means N, m and s. So here are the units inserted in the database through `editoast`:
- solid friction: N
- viscosity friction: N/(m/s)
- aerodynamic friction: N/(m/s)²

And finally, these values are combined with those of "towed rolling stocks" to produce a total rolling resistance to send for an STDCM request with the following formula:
- $`totalSolidFriction = tractionEngine[solidFriction] * tractionEngine[mass] + towed[solidFriction] * (towed[mass] - tractionEngine[mass])`$
- $`totalViscosityFriction = tractionEngine[viscosityFriction] * tractionEngine[mass] + towed[viscosityFriction] * (towed[mass] - tractionEngine[mass])`$
- $`totalAerodynamicFriction = tractionEngine[aerodynamicFriction] * tractionEngine[mass] + towed[aerodynamicFriction] * (towed[mass] - tractionEngine[mass])`$

⚠ If you followed all along, here are the problems $`tractionEngine[solidFriction]`$ is multiplied by $`tractionEngine[mass]`$ but it shouldn't because the former's unity is "N", not "N/kg". Same for $`towed[solidFriction]`$.

However, by an interesting hasard, the data of rolling resistance we have for towed rolling stocks have the following units:
- solid friction: daN/t
- viscosity friction: (daN/t)/(km/h)
- aerodynamic friction: (daN/t)/(km/h)²

So that means our formula are partly correct for the towed part.

⚠ But another problem is that we store both rolling resistance (for *rolling stocks* and for *towed rolling stocks*, so with different units) with the same `struct RollingResistance` in `editoast`. This `struct` has comment on which units are used but obviously, it is partly wrong because we store 2 different kind of units with the same `struct`.

After discussion with @flomonster, we will introduce a `struct RollingResistancePerWeight` for towed rolling stocks. And therefore, here are the formula we're going to use.

- $`totalSolidFriction = tractionEngine[solidFriction] + towed[solidFrictionPerWeight] * (towed[mass] - tractionEngine[mass])`$
- $`totalViscosityFriction = tractionEngine[viscosityFriction] + towed[viscosityFrictionPerWeight] * (towed[mass] - tractionEngine[mass])`$
- $`totalAerodynamicFriction = tractionEngine[aerodynamicFriction] + towed[aerodynamicFrictionPerWeight] * (towed[mass] - tractionEngine[mass])`$
- 
cc @axrolld : is the summary sound for you?

**Comment by axrolld** [comment]
Thanks so much for investigating ! I guess we didn't catch that mistake in the original PR...

The formulas you suggest at then end are correct for the units we are manipulating. The original formula was built for data per ton of RS so it was partly inapropriate.
